exchange/auction.go

```diff
@@ -91,53 +91,56 @@ func (a *auction) doCache(ctx context.Context, cache prebid_cache_client.Client,
 	for _, topBidsPerImp := range a.winningBidsByBidder {
 		for _, topBidPerBidder := range topBidsPerImp {
 			impID := topBidPerBidder.bid.ImpID
-			var customCacheKey string
-			var catDur string
-			useCustomCacheKey := false
-			if competitiveExclusion && topBidPerBidder == a.winningBids[impID] {
-				// set custom cache key for winning bid when competitive exclusion applies
-				catDur = bidCategory[topBidPerBidder.bid.ID]
-				if len(catDur) > 0 {
-					customCacheKey = fmt.Sprintf("%s_%s", catDur, hbCacheID)
-					useCustomCacheKey = true
-				}
-			}
-			if bids {
-				if jsonBytes, err := json.Marshal(topBidPerBidder.bid); err == nil {
-					if useCustomCacheKey {
-						// not allowed if bids is true; log error and cache normally
-						errs = append(errs, errors.New("cannot use custom cache key for non-vast bids"))
+			isOverallWinner := a.winningBids[impID] == topBidPerBidder
+			if includeBidderKeys || (includeWinners && isOverallWinner) {
```

Comment:
Again this can be shortened to `includeBidderKeys || isOverallWinner`. If `includeBidderKeys` is false, then we know `includeWinners` is true.

Reply:
You are right Hans, let me change this.

Reply:
changed